src/compose/ComposeMenu.js

```diff
@@ -63,72 +69,79 @@ export const chooseUploadImageFilename = (uri: string, fileName: ?string): strin
    * automatically converted to JPEG format... but the `fileName` property in
    * the react-native-image-picker response still has the `.HEIC` extension.
    */
+  // TODO: Is it still true, after the react-native-image-picker upgrade,
+  // that `fileName` can end in .HEIC?
   if (/\.jpe?g$/i.test(uri)) {
-    return name.replace(/\.heic$/i, '.jpeg');
+    const result = nameWithoutPrefix.replace(/\.heic$/i, '.jpeg');
+    if (result !== nameWithoutPrefix) {
+      logging.info('OK, so .HEIC to .jpeg replacement still seems like a good idea.');
```

Comment:
Looks good but I would not call this commit NFC. The new log event is a quite salient kind of behavior change for us as the developers (and we are after all the audience for the commit history :wink:), even if it's generally not noticeable for users.